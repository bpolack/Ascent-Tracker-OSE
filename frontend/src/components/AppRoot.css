/* Global App Styles & Variables */
.AscentApp {
	--ascent-pink: #fc5c7d;
	--ascent-blue: #6a82fb;
	--ascent-light: #f6f6f6;
	--ascent-dark: #343434;
	--ascent-text: #343434;
	--ascent-glass: rgba(255, 255, 255, 0.80);
	position: relative;
	background: var(--ascent-pink);
	background: -webkit-linear-gradient(115deg, var(--ascent-pink), var(--ascent-blue)); 
	background: linear-gradient(115deg, var(--ascent-pink), var(--ascent-blue));
	color: var(--ascent-text);
}
.AscentApp.Focus {
	--ascent-text: #f6f6f6;
	--ascent-glass: rgba(30, 30, 34, 0.95);
	background: var(--ascent-dark);
}
.LightHeading, .LightText {
	color: var(--ascent-light);
}
.LightHeading {
	text-shadow: 0px 0px 3px rgba(0,0,0,0.2);
}
.BigText {
	font-size: 120%;
}
.GlassBackground {
	background-color: var(--ascent-glass);
	backdrop-filter: blur( 4px );
    -webkit-backdrop-filter: blur( 4px );
}