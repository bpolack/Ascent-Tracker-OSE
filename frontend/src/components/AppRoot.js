import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Import App Components
import ParticleWrapper from './elements/ParticleWrapper/ParticleWrapper';
import FocusToggle from './elements/FocusToggle/FocusToggle';
import Header from './layout/Header/Header';
import Landing from './pages/Landing/Landing';
import Routes from './pages/Routes';
import AlertContainer from './layout/AlertContainer/AlertContainer';
import Footer from './layout/Footer/Footer';

export const AppRoot = (props) => {

	return (
		<Router>
			<div className={`${props.focus ? 'dark' : '' }`}>
				<div className="relative bg-gradient-to-tl from-pink-500 via-fuchsia-500 to-indigo-500 dark:from-indigo-900 dark:to-gray-800">
					<ParticleWrapper />
					<Header />
					<div className="relative">
						<FocusToggle />
						<Switch>
							<Route exact path="/" component={Landing} />
							<Route component={Routes} />
						</Switch>
					</div>
					<AlertContainer />
					<Footer />
				</div>
			</div>
		</Router>
	);
}

AppRoot.propTypes = {
	focus: PropTypes.bool.isRequired
}

const mapStateToProps = state => ({
	focus: state.focus
});

export default connect(mapStateToProps)(AppRoot)