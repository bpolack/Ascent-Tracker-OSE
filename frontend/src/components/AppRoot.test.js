import React from 'react';
import { shallow } from 'enzyme';
import { AppRoot } from './AppRoot';

describe('AppRoot', () => {
	it("Renders correctly without crashing", () => {
		const wrapper = shallow(<AppRoot focus={true} />);
		expect(wrapper).toMatchSnapshot();
	});
	it("Renders the correct classname for dark mode", () => {
		const wrapper = shallow(<AppRoot focus={true} />);
		expect(wrapper.find('div.dark').exists()).toBeTruthy();
	});
	it("Renders the correct classname for light mode", () => {
		const wrapper = shallow(<AppRoot focus={false} />);
		expect(wrapper.find('div.dark').exists()).toBeFalsy();
	});
});