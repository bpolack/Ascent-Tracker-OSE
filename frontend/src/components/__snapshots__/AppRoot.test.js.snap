// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoot Renders correctly without crashing 1`] = `ShallowWrapper {}`;
